### PR TITLE
Add uuid to table rows

### PR DESCRIPTION
**Pull request by fhlavac**
https://issues.redhat.com/browse/HAC-2127

this adds a uuid to each row data item if not defined

@florkbr

### CODE REVIEW

File: packages/lib-utils/package.json [hunk 1]
```diff
@@ -28,12 +28,14 @@
     "@patternfly/react-core": "^4.202.16",
     "@patternfly/react-table": "^4.71.16",
     "@patternfly/react-virtualized-extension": "^4.53.2",
+    "@types/uuid": "^8.3.4",
```
Comment (vojtechszocs) [inline packages/lib-utils/package.json:31]: We only need `uuid` in `peerDependencies`, no need for `@types/uuid` here.

Type only dependencies are typically development dependencies. We bundle any referenced vendor types in our own `d.ts` bundles.

Comment (fhlavac) [reply packages/lib-utils/package.json:31]: without the types the build is failing:
```
➤ YN0000: [!] (plugin typescript) Error: @rollup/plugin-typescript TS7016: Could not find a declaration file for module 'uuid'. '/go/src/github.com/openshift/dynamic-plugin-sdk/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
➤ YN0000:   Try `npm i --save-dev @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
➤ YN0000: src/components/table/VirtualizedTable.tsx (1:30)
➤ YN0000: 
➤ YN0000: 1 import { v4 as uuidv4 } from 'uuid';
➤ YN0000:                                ~~~~~~
```
do you think there is another way to fix it?

Comment (vojtechszocs) [reply packages/lib-utils/package.json:31]: @fhlavac AFAIK, we're using the `uuid` library as a helper to generate a fallback ID in `VirtualizedTable` component.

I'd put it in the same category as `lodash` or `immutable` i.e. put it into `dependencies`.

```diff
diff --git a/packages/lib-utils/package.json b/packages/lib-utils/package.json
index 6b217bd..a703f4b 100644
--- a/packages/lib-utils/package.json
+++ b/packages/lib-utils/package.json
@@ -28,19 +28,18 @@
     "@patternfly/react-core": "^4.202.16",
     "@patternfly/react-table": "^4.71.16",
     "@patternfly/react-virtualized-extension": "^4.53.2",
-    "@types/uuid": "^8.3.4",
     "react": "^17.0.2",
     "react-redux": "^7.2.2",
     "react-router-dom": "^6.3.0",
     "react-virtualized": "^9.22.3",
     "redux": "^4.1.2",
-    "redux-thunk": "^2.4.1",
-    "uuid": "^8.3.2"
+    "redux-thunk": "^2.4.1"
   },
   "dependencies": {
     "immutable": "^3.8.2",
     "lodash-es": "^4.17.21",
     "pluralize": "^8.0.0",
-    "typesafe-actions": "^4.4.2"
+    "typesafe-actions": "^4.4.2",
+    "uuid": "^8.3.2"
   }
 }
```

File: packages/lib-utils/package.json [hunk 2]
```diff
@@ -36,9 +36,11 @@
     "redux-thunk": "^2.4.1"
   },
   "dependencies": {
+    "@types/uuid": "^8.3.4",
```
Comment (vojtechszocs) [inline packages/lib-utils/package.json:39]: No need for type related dependencies here, any `uuid` related types we use in our library are automatically included in our library's `d.ts` bundle.

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

**Pull request merged**